$OpenBSD: patch-src_file_dl_posix_c,v 1.2 2021/10/16 17:57:34 rsadowski Exp $

Fix loading dlopen'd libraries on OpenBSD.
7bc73fd5cbd567ba128eec1f061323bed7a9c5d7

Index: src/file/dl_posix.c
--- src/file/dl_posix.c.orig
+++ src/file/dl_posix.c
@@ -85,11 +85,15 @@ void *dl_dlopen(const char *path, const char *version)
 #endif
 
     for (i = 0 ; search_paths[i] ; ++i) {
+#ifdef __OpenBSD__
+        name = str_printf("%s%s%s", search_paths[i], path, ext);
+#else
         if (version) {
             name = str_printf("%s%s%s.%s", search_paths[i], path, ext, version);
         } else {
             name = str_printf("%s%s%s", search_paths[i], path, ext);
         }
+#endif
 
         if (!name) {
             BD_DEBUG(DBG_FILE | DBG_CRIT, "out of memory\n");
